core-test/src/test/scala/org/bitcoins/core/number/BasicArithmeticSpec.scala

```diff
@@ -0,0 +1,63 @@
+package org.bitcoins.core.number
+import org.bitcoins.core.gen.NumberGenerator
+import org.bitcoins.core.util.BitcoinSUnitTest
+import org.scalatest.prop.PropertyChecks
+
+class BasicArithmeticSpec extends BitcoinSUnitTest {
+
+  // We have to wrap BasicArithmetic instead of doing
+  // an anonymous class, that causes overloading confusion
+  private case class NumWrapper(underlying: BigInt)
+      extends BasicArithmetic[NumWrapper] {
+
+    override def +(n: NumWrapper): NumWrapper =
+      NumWrapper(underlying + n.underlying)
+
+    override def -(n: NumWrapper): NumWrapper =
+      NumWrapper(underlying - n.underlying)
+
+    override def *(factor: BigInt): NumWrapper =
+      NumWrapper(underlying * factor)
+
+    override def *(factor: NumWrapper): NumWrapper =
+      NumWrapper(underlying * factor.underlying)
+  }
+
+  private val numWrapperGen = for {
+    int <- NumberGenerator.bigInts
+  } yield NumWrapper(int)
+
+  behavior of "BasicArithmetic"
+
+  it must "multiply safely and unsafely with an int" in {
+    PropertyChecks.forAll(NumberGenerator.bigInts, numWrapperGen) { (i, num) =>
+      val unsafe = num * i
+      val safe = num.multiplySafe(i)
+      assert(safe.toOption.contains(unsafe))
```

Comment:
I prefer 

```scala
safe.get == unsafe
```
It is just tests after all, so i'm not as concerned about being type safe. Worst case is it throws a `NoSuchElementException`.